tests/tests_ut/az_ulib_ipc_ut/az_ulib_ipc_ut.c

```diff
@@ -793,64 +1073,391 @@ static void az_ulib_ipc_publish_return_handle_succeed(void** state)
   assert_int_equal(out, AZ_OK);
 
   /// cleanup
-  assert_int_equal(az_ulib_ipc_release_interface(interface_handle_copy), AZ_OK);
-  assert_int_equal(az_ulib_test_my_interface_1_v123_unpublish(AZ_ULIB_NO_WAIT), AZ_OK);
-  assert_int_equal(az_ulib_test_my_interface_1_v2_unpublish(AZ_ULIB_NO_WAIT), AZ_OK);
-  assert_int_equal(az_ulib_test_my_interface_2_v123_unpublish(AZ_ULIB_NO_WAIT), AZ_OK);
-  assert_int_equal(az_ulib_test_my_interface_3_v123_unpublish(AZ_ULIB_NO_WAIT), AZ_OK);
-  assert_int_equal(az_ulib_ipc_deinit(), AZ_OK);
+  assert_int_equal(az_ulib_ipc_release_interface(interface_handle_copy), AZ_OK);
+  assert_int_equal(az_ulib_test_my_interface_a_1_1_123_unpublish(AZ_ULIB_NO_WAIT), AZ_OK);
+  assert_int_equal(az_ulib_test_my_interface_a_1_1_200_unpublish(AZ_ULIB_NO_WAIT), AZ_OK);
+  assert_int_equal(az_ulib_test_my_interface_a_1_2_123_unpublish(AZ_ULIB_NO_WAIT), AZ_OK);
+  assert_int_equal(az_ulib_test_my_interface_a_1_3_123_unpublish(AZ_ULIB_NO_WAIT), AZ_OK);
+  assert_int_equal(az_ulib_ipc_deinit(), AZ_OK);
+}
+
+/* If the provided package version is AZ_ULIB_VERSION_DEFAULT, the az_ulib_ipc_publish shall return
```

Comment:
>AZ_ULIB_VERSION_DEFAULT [](http://example.com/codeflow?start=38&length=23)

is this because you can only set default after publishing with an explicit version number? I ask this because I know packages are allowed to have a default version, so just double checking

Reply:
When you publish an interface, you must specify the package version of that interface. "Default" is only to get a interface.